test/test_site/testUtil/diffHtml.js

```diff
@@ -25,12 +25,16 @@ const endsWithUnclosedPath = (fragment) => {
  *        <div/>
  */
 const endsWithOpeningTag = (fragment) => {
+  let numUnmatchedClosingBracket = 0;
   for (let i = fragment.length - 1; i >= 0; i -= 1) {
     if (fragment[i] === '<') {
-      return true;
+      numUnmatchedClosingBracket -= 1;
+      if (numUnmatchedClosingBracket < 0) {
+        return true;
+      }
     }
     if (fragment[i] === '>') {
-      return false;
+      numUnmatchedClosingBracket += 1;
```

Comment:
Maybe more intuitive to +1 before -1, and check for 0 explicitly.

```
if > {
  +1
} else if < {
  if 0 {
    return true
  }
  -1
}
```

Reply:
I've updated the commit